### PR TITLE
modification of the schema.json

### LINKED ISSUE #240: revise the schema.json
The schema file needed to be revised to make sure all the design parameters are matching the modelica files

https://github.com/urbanopt/geojson-modelica-translator/blob/develop/geojson_modelica_translator/system_parameters/schema.json

### PR DESCRIPTION
**Pull request by macintoshpie**
Note: this is a clone of PR #241 

The system parameter schema.json needed to be reviewed and updated to reflect the latest modelica models.

The PR resolves #240 where 

- Buildings parameters have been updated to include T heating/cooling air set_point, Chilled/heating water supply and return temperature for building and district level.

- Removed  a non required old parameter section" central plant".

### CODE REVIEW

File: geojson_modelica_translator/system_parameters/schema.json [hunk 1]
```diff
@@ -291,91 +269,137 @@
...
-        "nominal_flow_district",
-        "nominal_flow_building",
-        "pressure_drop_valve",
-        "pressure_drop_hx_secondary",
-        "pressure_drop_hx_primary",
-        "cooling_supply_water_temperature_district",
-        "cooling_supply_water_temperature_building",
-        "heating_supply_water_temperature_district",
-        "heating_supply_water_temperature_building"
+        "Q_flow_nominal",
```
Comment (macintoshpie) [inline geojson_modelica_translator/system_parameters/schema.json:352]: No such property, should this be `heat_flow_nominal`?

File: geojson_modelica_translator/system_parameters/schema.json [hunk 2]
```diff
@@ -291,91 +269,137 @@
...
+        "hx_efficiency",
+        "nominal_mass_flow_district",
+        "nominal_mass_flow_building",
+        "valve_pressure_drop",
+        "hx_secondary_pressure_drop",
+        "hx_primary_pressure_drop",
+        "temp_district_chw_supply",
+        "temp_district_chw_supply",
```
Comment (macintoshpie) [inline geojson_modelica_translator/system_parameters/schema.json:360]:
```suggestion
```
duplicate. Or should this be `temp_building_chw_supply`?

File: geojson_modelica_translator/system_parameters/schema.json [hunk 3]
```diff
@@ -429,92 +459,264 @@
...
+          "type": "number",
+          "default": 298.15
+        },
+        "temp_cw_in_nominal": {
+          "description": "Nominal cooling water inlet temperature. (K)",
+          "type": "number",
+          "default": 308.15
+        },
+        "cooling_tower_water_temperature_difference_nominal": {
+          "Nominal water temperature difference between of the tower. (K)",
```
Comment (macintoshpie) [inline geojson_modelica_translator/system_parameters/schema.json:636]:
```suggestion
          "description": "Nominal water temperature difference between of the tower. (K)",
```

File: geojson_modelica_translator/system_parameters/schema.json [hunk 4]
```diff
@@ -429,92 +459,264 @@
...
+          "type": "number",
+          "default": 280.15
+        },
+        "delta_temp_approach": {
+          "Approach tempearture difference. (K)",
```
Comment (macintoshpie) [inline geojson_modelica_translator/system_parameters/schema.json:641]:
```suggestion
          "description": "Approach tempearture difference. (K)",
```

File: geojson_modelica_translator/system_parameters/schema.json [hunk 5]
```diff
@@ -429,92 +459,264 @@
...
+          "type": "number",
+          "default": 3
+        },
+        "ratio_water_air_nominal": {
+          "water_air_ratio_nominal",
```
Comment (macintoshpie) [inline geojson_modelica_translator/system_parameters/schema.json:646]:
```suggestion
          "description": "water_air_ratio_nominal",
```

File: geojson_modelica_translator/system_parameters/schema.json [hunk 6]
```diff
@@ -429,92 +459,264 @@
...
+          "type": "number",
+          "default": 0.625
+        },
```
Comment (macintoshpie) [inline geojson_modelica_translator/system_parameters/schema.json:649]:
```suggestion
        }
```

File: geojson_modelica_translator/system_parameters/schema.json [hunk 7]
```diff
@@ -429,92 +459,264 @@
...
+        "temp_setpoint_hw": {
+          "description": "District circuit heating water temperature setpoint. (K)",
+          "type": "number",
+          "default": 338.15
+        },
+        "pressure_drop_hw_valve_nominal": {
+          "description": "Chiller isolation valve pressure drop. (Pa)",
+          "type": "number",
+          "default": 6000
+        },
```
Comment (macintoshpie) [inline geojson_modelica_translator/system_parameters/schema.json:709]:
```suggestion
        }
```
JSON doesn't allow trailing commas

**Review by macintoshpie — COMMENTED** [review]
Some suggested syntax fixes, note that I did not check the validity of the schema changes themselves

**Comment by nllong** [comment]
I think this is really close. @macintoshpie / @Helarga  can you review again? I don't think we should merge this until we fix the tests because they would probably catch any of the items that we missed.

File: geojson_modelica_translator/model_connectors/energy_transfer_systems/templates/CoolingIndirect.mot [hunk 8]
```diff
@@ -12,11 +12,11 @@ model CoolingIndirect
   // mass flow rates
   parameter Modelica.SIunits.MassFlowRate mDis_flow_nominal(
     final min=0,
-    {% endraw %}start={{ets_data["nominal_flow_district"]}})
+    {% endraw %}start={{ets_data["valve_pressure_drop"]}})
```
Comment (nllong) [inline geojson_modelica_translator/model_connectors/energy_transfer_systems/templates/CoolingIndirect.mot:15]:
```suggestion
    {% endraw %}start={{ets_data["nominal_flow_district"]}})
```

File: geojson_modelica_translator/model_connectors/energy_transfer_systems/templates/HeatingIndirect.mot [hunk 9]
```diff
@@ -12,11 +12,11 @@ model HeatingIndirect
   // mass flow rates
   parameter Modelica.SIunits.MassFlowRate mDis_flow_nominal(
     final min=0,
-    {% endraw %}start={{ets_data["nominal_flow_district"]}})
+    {% endraw %}start={{ets_data["valve_pressure_drop"]}})
```
Comment (nllong) [inline geojson_modelica_translator/model_connectors/energy_transfer_systems/templates/HeatingIndirect.mot:15]:
```suggestion
    {% endraw %}start={{ets_data["nominal_mass_flow_building"]}})
```

**Comment by nllong** [comment]
> I think this is really close. @macintoshpie / @Helarga can you review again? I don't think we should merge this until we fix the tests because they would probably catch any of the items that we missed.

I've noticed some test failures. I'll look into those. Feel free to review knowing that the system param files are not valid (yet).

File: tests/system_parameters/data/system_params_1.json [hunk 10]
```diff
@@ -11,19 +11,18 @@
...
+          "nominal_mass_flow_building": 0.5,
+          "valve_pressure_drop": 6000,
+          "heat_exchanger_secondary_pressure_drop": 500,
+          "heat_exchanger_primary_pressure_drop": 500,
           "cooling_supply_water_temperature_district": 5,
           "cooling_supply_water_temperature_building": 7,
           "heating_supply_water_temperature_district": 55,
           "heating_supply_water_temperature_building": 50,
           "booster_heater": false,
-          "ets_generation": "Fifth Generation",
```
Comment (Helarga) [inline tests/system_parameters/data/system_params_1.json:26]: @nllong are the 5thG ETS parameters included at the schema.json or other file? also these parameters        
 "primary_design_delta_t": 3,
 "secondary_design_delta_t": 3
i am not sure where i can find it.

File: tests/system_parameters/data/system_params_2.json [hunk 11]
```diff
@@ -11,19 +11,18 @@
...
+          "nominal_mass_flow_building": 0.5,
+          "valve_pressure_drop": 6000,
+          "heat_exchanger_secondary_pressure_drop": 500,
+          "heat_exchanger_primary_pressure_drop": 500,
           "cooling_supply_water_temperature_district": 5,
           "cooling_supply_water_temperature_building": 7,
           "heating_supply_water_temperature_district": 80,
           "heating_supply_water_temperature_building": 75,
           "booster_heater": false,
-          "ets_generation": "Fifth Generation",
```
Comment (Helarga) [inline tests/system_parameters/data/system_params_2.json:26]: Fifth Generation type confirm

Comment (nllong) [reply tests/system_parameters/data/system_params_2.json:26]: yeah, it is just for testing purposes.

File: tests/system_parameters/data/system_params_1.json (hunk 10 above)
Comment (nllong) [reply tests/system_parameters/data/system_params_1.json:26]: right now the parameters are the same. If you want to propose breaking up the parameters by generation, then feel free to do that. But I would wait until we get through this first.

**Comment by vtnate** [comment]
As of this now the only failing tests are `tests/model_connectors/test_district_heating_and_cooling_systems.py` and `tests/model_connectors/test_district_heating_system.py`. The `...results/stdout` files don't show the same error message, but it seems likely that they are caused by the same issue, whatever it is.

This PR now includes the work from #267 for testing. Proper solution may be to revert that commit in this PR once #267 has been merged to develop, and merge develop into this PR at that time.

File: geojson_modelica_translator/model_connectors/plants/cooling_plant.py [hunk 12]
```diff
@@ -57,22 +57,58 @@ def to_modelica(self, scaffold):
...
                 "fan_power": self.system_parameters.get_param(
-                    "$.district_system.default.cooling_plant.fan_power_nominal"
+                    "$.district_system.default.central_cooling_plant_parameters.cooling_tower_fan_power_nominal"
                 ),
-                "chilled_water_pump_pressure_drop": self.system_parameters.get_param(
-                    "$.district_system.default.cooling_plant.chilled_water_pump_pressure_drop"
+                "chilled_water_pump_pressure_head": self.system_parameters.get_param(
+                    "$.district_system.default.central_cooling_plant_parameters.chw_pump_head"
+                ),
+                "condenser_water_pump_pressure_head": self.system_parameters.get_param(
```
Comment (nllong) [inline geojson_modelica_translator/model_connectors/plants/cooling_plant.py:73]: nice! Tanks!

**Review by nllong — APPROVED** [review]
This looks good. Thanks everyone for getting this across the finish line.

**Pull request merged**